test/e2e/webhook_reserved_namespace_resources_test.go

```diff
@@ -0,0 +1,1350 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package e2e
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+
+	"go.goms.io/fleet/pkg/utils"
+	testutils "go.goms.io/fleet/test/e2e/utils"
+)
+
+const (
+	testRole        = "test-role"
+	testRoleBinding = "test-role-binding"
+	testPod         = "test-pod"
+	testService     = "test-service"
+	testSecret      = "test-secret"
+	testDaemonSet   = "test-daemon-set"
+	testDeployment  = "test-deployment"
+	testReplicaSet  = "test-replica-set"
+	testConfigMap   = "test-config-map"
+	testCronJob     = "test-cron-job"
+	testJob         = "test-job"
+)
+
+var _ = Describe("Fleet's Reserved Namespaced Resources Handler webhook tests", func() {
+	Context("fleet guard rail e2e for role", func() {
+		BeforeEach(func() {
+			rName := testRole + fmt.Sprint(GinkgoParallelProcess())
+			r := rbacv1.Role{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      rName,
+					Namespace: memberNamespace.Name,
+				},
+				Rules: []rbacv1.PolicyRule{
+					{
+						Verbs:     []string{"*"},
+						APIGroups: []string{"*"},
+						Resources: []string{"*"},
+					},
+				},
+			}
+			Expect(HubCluster.KubeClient.Create(ctx, &r)).Should(Succeed())
+		})
+
+		AfterEach(func() {
+			rName := testRole + fmt.Sprint(GinkgoParallelProcess())
```

Comment:
why are you deleting a Role that is different from the one that is setup in the Before each?